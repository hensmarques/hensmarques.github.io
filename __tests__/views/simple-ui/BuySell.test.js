import React from 'react';
import { BuySell } from '../../../src/views/simple-ui/BuySell';

describe('<BuySell/>', () => {
  const mockProps = {
    instruments: [],
    selectedInstrument: {
      id: 'BTCUSD',
      base: 'BTC',
      quote: 'USD'
    },
    selectedBidAsk: {
      ask: 3000,
      bid: 3030
    },
    changeInstrument: jest.fn()
  };

  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<BuySell {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the instrument options in the select dropdown', () => {
    const instruments = [{ id: 'BTCJPY' }, { id: 'BTCGBP' }, { id: 'XRPUSD' }];
    mockProps.instruments = instruments;
    const wrapper = shallow(<BuySell {...mockProps} />);
    const selectOptions = wrapper.find('Select').props().options;
    expect(selectOptions[0].name).toBe(instruments[0].id);
    expect(selectOptions[0].value).toBe(instruments[0].id);
    expect(selectOptions[1].name).toBe(instruments[1].id);
    expect(selectOptions[1].value).toBe(instruments[1].id);
    expect(selectOptions[2].name).toBe(instruments[2].id);
    expect(selectOptions[2].value).toBe(instruments[2].id);
  });

  it('dispatches changeInstrument when dropdown select value changes', () => {
    const mockEvent = { target: { value: 'BTCJPY' } };
    const wrapper = shallow(<BuySell {...mockProps} />);
    wrapper.instance().onInstrumentChange(mockEvent);
    expect(mockProps.changeInstrument).toHaveBeenCalledWith(mockEvent.target.value);
  });

  it('switches the BuySellSelector value when transaction type changes', () => {
    const wrapper = shallow(<BuySell {...mockProps} />);
    expect(wrapper.find('BuySellSelector').props().value).toBe('buy');
    wrapper.setState({ transactionType: 'sell' });
    expect(wrapper.find('BuySellSelector').props().value).toBe('sell');
  });
});
