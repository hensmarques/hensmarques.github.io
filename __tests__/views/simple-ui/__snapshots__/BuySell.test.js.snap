// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuySell/> renders and matches the snapshot 1`] = `
<Connect(SimpleUiWrapper)
  description="Select the amount you want to buy or sell, or click the custom button to buy or sell a different amount."
  pageClass="buy-sell"
  pageHeader="As easy as 1-2-3"
  title="Buy/Sell"
>
  <div
    className="ui-controls"
  >
    <BuySellSelector
      currencyName="BTC"
      onChange={[Function]}
      value="buy"
    />
    <Select
      className=""
      hideLabel={true}
      name="instrument"
      onChange={[Function]}
      options={Array []}
      placeholder=""
      translation="BUY_SELL"
      value="BTCUSD"
      writeOnce={false}
    />
  </div>
  <div
    className="buy-sell-option-container"
  >
    <BuySellOption
      amount={100}
      baseCurrency="BTC"
      key="0"
      quantity={0.03333333}
      quoteCurrency="USD"
      t="BUY_SELL"
      transactionType="buy"
    />
    <BuySellOption
      amount={200}
      baseCurrency="BTC"
      key="1"
      quantity={0.06666666}
      quoteCurrency="USD"
      t="BUY_SELL"
      transactionType="buy"
    />
    <BuySellOption
      amount={500}
      baseCurrency="BTC"
      key="2"
      quantity={0.16666666}
      quoteCurrency="USD"
      t="BUY_SELL"
      transactionType="buy"
    />
    <BuySellOption
      amount={1000}
      baseCurrency="BTC"
      key="3"
      quantity={0.33333333}
      quoteCurrency="USD"
      t="BUY_SELL"
      transactionType="buy"
    />
  </div>
  <div
    className="custom-amount-container"
  >
    <p>
      <Link
        replace={false}
        to="/buy-sell-custom"
      >
        Click here
      </Link>
       
      to input a custom amount of digital currency.
    </p>
  </div>
</Connect(SimpleUiWrapper)>
`;
